extensions/ext/GLSL_EXT_integer_dot_product.txt

```diff
@@ -0,0 +1,287 @@
+
+      uint dotAccSatEXT(uvec2 a, uvec2 b, uint c)
+      int dotAccSatEXT(ivec2 a, uvec2 b, int c)
+      int dotAccSatEXT(uvec2 a, ivec2 b, int c)
+      int dotAccSatEXT(ivec2 a, ivec2 b, int c)
+
+      uint dotAccSatEXT(uvec3 a, uvec3 b, uint c)
+      int dotAccSatEXT(ivec3 a, uvec3 b, int c)
+      int dotAccSatEXT(uvec3 a, ivec3 b, int c)
+      int dotAccSatEXT(ivec3 a, ivec3 b, int c)
+
+      uint dotAccSatEXT(uvec4 a, uvec4 b, uint c)
+      int dotAccSatEXT(ivec4 a, uvec4 b, int c)
+      int dotAccSatEXT(uvec4 a, ivec4 b, int c)
+      int dotAccSatEXT(ivec4 a, ivec4 b, int c)
+
+      uint dotPacked4x8AccSatEXT(uint a, uint b, uint c)
+      int dotPacked4x8AccSatEXT(int a, uint b, int c)
+      int dotPacked4x8AccSatEXT(uint a, int b, int c)
+      int dotPacked4x8AccSatEXT(int a, int b, int c)
+
+      If GL_EXT_shader_explicit_arithmetic_types_int8 is enabled:
+
+      uint dotAccSatEXT(u8vec2 a, u8vec2 b, uint c)
+      int dotAccSatEXT(i8vec2 a, u8vec2 b, int c)
+      int dotAccSatEXT(u8vec2 a, i8vec2 b, int c)
+      int dotAccSatEXT(i8vec2 a, i8vec2 b, int c)
+
+      uint dotAccSatEXT(u8vec3 a, u8vec3 b, uint c)
+      int dotAccSatEXT(i8vec3 a, u8vec3 b, int c)
+      int dotAccSatEXT(u8vec3 a, i8vec3 b, int c)
+      int dotAccSatEXT(i8vec3 a, i8vec3 b, int c)
+
+      uint dotAccSatEXT(u8vec4 a, u8vec4 b, uint c)
+      int dotAccSatEXT(i8vec4 a, u8vec4 b, int c)
+      int dotAccSatEXT(u8vec4 a, i8vec4 b, int c)
+      int dotAccSatEXT(i8vec4 a, i8vec4 b, int c)
+
+      If GL_EXT_shader_explicit_arithmetic_types_int16 is enabled:
+
+      uint dotAccSatEXT(u16vec2 a, u16vec2 b, uint c)
+      int dotAccSatEXT(i16vec2 a, u16vec2 b, int c)
+      int dotAccSatEXT(u16vec2 a, i16vec2 b, int c)
+      int dotAccSatEXT(i16vec2 a, i16vec2 b, int c)
+
+      uint dotAccSatEXT(u16vec3 a, u16vec3 b, uint c)
+      int dotAccSatEXT(i16vec3 a, u16vec3 b, int c)
+      int dotAccSatEXT(u16vec3 a, i16vec3 b, int c)
+      int dotAccSatEXT(i16vec3 a, i16vec3 b, int c)
+
+      uint dotAccSatEXT(u16vec4 a, u16vec4 b, uint c)
+      int dotAccSatEXT(i16vec4 a, u16vec4 b, int c)
+      int dotAccSatEXT(u16vec4 a, i16vec4 b, int c)
+      int dotAccSatEXT(i16vec4 a, i16vec4 b, int c)
+
+      If GL_EXT_shader_explicit_arithmetic_types_int64 is enabled:
+
+      uint64_t dotAccSatEXT(u64vec2 a, u64vec2 b, uint c)
+      int64_t dotAccSatEXT(i64vec2 a, u64vec2 b, int c)
+      int64_t dotAccSatEXT(u64vec2 a, i64vec2 b, int c)
+      int64_t dotAccSatEXT(i64vec2 a, i64vec2 b, int c)
+
+      uint64_t dotAccSatEXT(u64vec3 a, u64vec3 b, uint c)
+      int64_t dotAccSatEXT(i64vec3 a, u64vec3 b, int c)
+      int64_t dotAccSatEXT(u64vec3 a, i64vec3 b, int c)
+      int64_t dotAccSatEXT(i64vec3 a, i64vec3 b, int c)
+
+      uint64_t dotAccSatEXT(u64vec4 a, u64vec4 b, uint c)
+      int64_t dotAccSatEXT(i64vec4 a, u64vec4 b, int c)
+      int64_t dotAccSatEXT(u64vec4 a, i64vec4 b, int c)
+      int64_t dotAccSatEXT(i64vec4 a, i64vec4 b, int c)
```

Comment:
The accumulator `c` should be 64bit too for these.

Reply:
Good catch, thanks! Now fixed. (I've added you to the list of contributors but couldn't find your affiliation. Feel free to comment with a suggestion if you want one added.)